src/main/java/pl/wedding/management/model/reservation/ReservationDto.java

```diff
@@ -0,0 +1,34 @@
+package pl.wedding.management.model.reservation;
+
+import lombok.*;
+
+import java.time.LocalDateTime;
+
+@Data
+@Builder
+@AllArgsConstructor
+public class ReservationDto {
+
+    private final long id;
+    private final LocalDateTime reservationStart;
+    private final LocalDateTime reservationEnd;
+    private final Object weddingHall; //to be changed
+    private final Object tenant; //to be changed
+    private final int numberOfGuests;
+    private final Occasion occasion;
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Z tego co widzę to nie jest Encja bazodanowa. W takim razie, ja bym nie szedł tą drogą. Taki equals i hashCode są ok, dla Hibernate, dla którego jedynym wyznacznikiem czy dwie encje są tym samym jest właśnie ID. 

Tu mamy DTO i wyobraź sobie że dwa obiekty które mają inną datę rezerwacji, mimo iż mają to samo ID to nie są te same obiekty (przechowywane np. w set-ach). To że mają to samo ID, oznacza tylko tyle że dane się zmieniły i trzeba je zaktualizować/persystować, ale dla wydajności setów, map, list, tablic ważne jest by mieć możliwość precyzyjnego określenia czy obiekty są takie same. Zostawiłbym implementację lomboka.

Reply:
Equals jest jednym z bardziej ślizgawych tematów jeśli chodzi o naszą kochaną Javę. Często poruszany przez Jarka Ratajeskiego na konferencjach jako jeden z najgorszych elementów naszego języka. Nie ma ścisłej konwencji jak powinniśmy obsługiwać ten equals - czy w Twój sposób porównując wszystkie pola czy w mój sposób - porównując samo ID.

Nie zgodzę się, że rezerwacją mająca to samo ID ale inną datę powinna się zaktualizować w bazie danych. W takim przypadku to są dwie różne rezerwacje i tak powinny być traktowane. Taka sytuacja w ogóle nie powinna mieć miejsca.